app/controllers/media_controller.rb

```diff
@@ -0,0 +1,54 @@
+class MediaController < ApplicationController
+  before_action only: [:show, :edit, :update, :destroy, :vote] { @medium = Medium.find(params[:id]) }
+  before_action only: [ :edit, :destroy, :vote] { @type = @medium.type.pluralize }
+
+  def index
+    @media = Medium.where("type = '#{params[:type]}'")
+  end
+
+  def show; end
+
+  def new
+    @medium = Medium.new
+    @type = params[:type]
+  end
+
+  def create
+    @medium = Medium.create(media_params[:medium])
+
+    if @medium.save
+      @media_type = params[:medium][:type]
+      redirect_to polymorphic_path(@medium)
+    else
+      render "new"
+    end
+  end
+
+  def edit; end
+
+  def update
+    media_type = @medium.type.downcase.singularize
+
+    @medium.update(
+      creator: params[media_type][:creator], 
+      name: params[media_type][:name], 
+      description: params[media_type][:description])
```

Comment:
This update method could be written as `@medium.update(media_params)`. The way the media_params (strong parameters) are currently written it's not possible to do it this way but if you require the media type (book, album, etc) and permit the params, this would help eliminate a few lines of code.